File: HackerRank/Algorithm/WarmUp/cut_the_sticks.cc
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

// https://www.hackerrank.com/challenges/cut-the-sticks

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */ 
    int num = 0;
    cin >> num;
    vector<int> input;
    while(num > 0) {
    	num--;
    	int i;
    	cin >> i;
    	input.push_back(i);
    }

    bool flag = true;
    while(flag) {
    	int result = *min_element(begin(input), std::end(input));
	    int count = 0;

	    for(vector<int>::iterator it = input.begin(); it != input.end(); it++) {
    		if(*it != 0) {
	    		count++;
	    		*it -= result;
	    	}
	    }

	    input.erase(remove(input.begin(), input.end(), 0), input.end());
	    flag = count != 0;

	    if(count == 0) {
	    	break;
	    }
	    // cout << "\n=========" << "\n" << count << "\n=========" << "\n";
	    // for(int x : input) {
    	// 	cout << x << "\n";
    	// }
    	// break;

    	cout << count << "\n";
	}

    return 0;
}

File: HackerRank/Algorithm/WarmUp/cut_the_sticks/cut_the_sticks.py
#!/usr/bin/python

# https://www.hackerrank.com/challenges/cut-the-sticks

import fileinput

def main():
    input = fileinput.input()
    num = int(input.readline().strip())
    out = [int(i) for i in input.readline().strip().split(' ')]

    flag = True
    while(flag):
        count = len(out)
        minimum = min(out)
        reduction = 0
        for i in range(count):
            out[i] -= minimum
            if(out[i] == 0):
                reduction += 1

        for i in range(reduction):
            out.remove(0)

        print count
        flag = len(out) > 0
        
if __name__ == '__main__':
    main()
